### PR TITLE
List documents

### PR DESCRIPTION
**Pull request by FrozenKiwi**
# Description

Implements `listDocuments`, and fixes the implementation of `CollectionReference.get` to match Firestore

## Related issues

None

## How to test

Tests included (all existing tests pass without changes.

NOTE: This is -kinda- a breaking change, as the previous implementation of `CollectionReference.get` would return empty document references.  This isn't a common pattern, and if you want to preserve the original behaviour I think the best choice would be to revert that part of the changelist (rather than putting the change behind a flag etc).

### CODE REVIEW

**Comment by FrozenKiwi** [comment]
note, I'll rebase the next PR off master to avoid confusing git with that history

File: mocks/firestore.js [hunk 1]
```diff
@@ -464,12 +467,14 @@ FakeFirestore.CollectionReference = class extends FakeFirestore.Query {
...
-    const records = this._records();
-    records.forEach(rec => {
-      rec._ref = new FakeFirestore.DocumentReference(rec.id, this, this.firestore);
-    });
+    const records = this._records().map(rec => ({
+      ...rec,
+      _ref: new FakeFirestore.DocumentReference(rec.id, this, this.firestore),
+    }));
+    // Firestore does not return documents with no local data
+    const onlyLocal = records.filter(r => Object.keys(r._ref._get().data()).length > 0);
```
Comment (sbatson5) [inline mocks/firestore.js:475]: Can we actually get this in a separate PR? It feels unrelated to the `listDocuments` change and in case this _does_ introduce something that is breaking, we can roll it back without losing the `listDocument` changes

**Review by sbatson5 — CHANGES_REQUESTED** [review]

File: mocks/firestore.js (hunk 1 above)
Comment (FrozenKiwi) [reply mocks/firestore.js]: a year or so later - it has been removed

**Comment by FrozenKiwi** [comment]
Hey @sbatson5 - I know it took a while (a whole year???) but I've made the changes you requested and updated to latest.  Any chance we could get this in?

**Review by sbatson5 — APPROVED** [review]
